pkg/controller/controllerutil/controllerutil_test.go

```diff
@@ -44,10 +48,94 @@ var _ = Describe("Controllerutil", func() {
 			Expect(controllerutil.SetControllerReference(&errMetaObj{}, rs, scheme.Scheme)).To(HaveOccurred())
 		})
 	})
+
+	Describe("CreateOrUpdate", func() {
+
+		It("creates a new object if one doesn't exists", func() {
+			deplKey := types.NamespacedName{Name: "test-create", Namespace: "default"}
+			depl := &appsv1.Deployment{}
+
+			op, err := controllerutil.CreateOrUpdate(context.TODO(), c, deplKey, depl, createDeployment)
+
+			By("returning OperationCreated")
+			Expect(op).Should(BeEquivalentTo(controllerutil.OperationCreated))
+
+			By("returning returning no error")
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("actually having the deployment created")
+			fetched := &appsv1.Deployment{}
+			Expect(c.Get(context.TODO(), deplKey, fetched)).Should(Succeed())
+		})
+
+		It("update existing object", func() {
+			deplKey := types.NamespacedName{Name: "test-update", Namespace: "default"}
+			d, _ := createDeployment(&appsv1.Deployment{})
+			depl := d.(*appsv1.Deployment)
+			depl.Name = "test-update"
+			depl.Namespace = "default"
+
+			var scale int32 = 2
+
+			Expect(c.Create(context.TODO(), depl)).Should(Succeed())
```

Comment:
style is general `Expect(X).To(Y)` (`Should` is reserved for the omega character, which we don't use)